hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -477,14 +479,19 @@ private Pair<SchemaProvider, Pair<String, JavaRDD<HoodieRecord>>> fetchFromSourc
     }
 
     boolean shouldCombine = cfg.filterDupes || cfg.operation.equals(WriteOperationType.UPSERT);
+    boolean isDropPartitionColumns = props.getBoolean(DROP_PARTITION_COLUMNS.key());
+    String partitionColumns = HoodieSparkUtils.getPartitionColumns(keyGenerator, props);
+    List<String> listOfPartitionColumns = Arrays.asList(partitionColumns.split(","));
```

Comment:
For a non-partitioned table, will this be empty list or a list of single element that is empty string?

Reply:
Good question, It would be an empty list, I've added a UT for a non-partitioned table.